Tutorial.md

```diff
@@ -11,477 +12,247 @@ Gist is a paste. A snippet of text. Not unrelated to [GitHub Gists][].  To set s
 
-In API Blueprint a resource definition starts with a Markdown header with the resource name followed by its URI enclosed in square brackets. The URI is relative to the API root. So in this case it is just `/`.
+## Resource Groups
 
-> **NOTE:** Resource name is an [API Blueprint Identifier][] and as such it can be composed only from a combination of alphanumerical characters, underscores, dashes and a spaces.
+Now it's time to start documenting the resources inside of the API. To start of with, the `Group` keyword is used inside a heading to specify a group of related resources.
 
-#### Resource Description
+```markdown
+# Group Questions
 
-    # Gist Fox API Root [/]
-    Gist Fox API entry point.
+Resource related to questions in the API.
+```
 
-    This resource does not have any attributes. Instead it offers the initial API affordances in the form of the HTTP Link header and haveL links.
+## Resource
 
-As with the API Name the resource name can be followed by an arbitrary Markdown formatted description.
+Inside of the questions resource group, we have a resource called "Questions collection". This resource allows you to view a list of questions. The heading specifies the URI used to access the resource inside of square brackets at the end of the heading.
 
-#### Action
+```markdown
+## Questions collection [/questions]
+```
 
-    ## Retrieve Entry Point [GET]
+### Actions
 
-Here we have just defined an action named "Retrieve Entry Point" utilizing the `GET` [HTTP Request Method][]. As with a resource and API name we can add any arbitrary description here but since the action name is pretty self explanatory we will skip it for now.
+API Blueprint allows you to specify the actions you can make on each resource. A sub-heading inside of the resource with a title, followed by the HTTP method to use inside the square brackets.
 
-#### Response
+```markdown
+### List all questions [GET]
+```
 
-    + Response 200
-
-In API Blueprint an action **should** always include at least one response representing the HTTP response message sent back in response to a HTTP request. The response should always bear a [status code][] and possibly an additional [payload][]. Here we have defined the most common response "200" indicating the request has succeeded.
-
-#### Response Payload
-
-    + Response 200 (application/hal+json)
-        + Headers
-
-                Link: <http:/api.gistfox.com/>;rel="self",<http:/api.gistfox.com/gists>;rel="gists"
-
-        + Body
-
-                {
-                    "_links": {
-                        "self": { "href": "/" },
-                        "gists": { "href": "/gists?{since}", "templated": true }
-                    }
-                }
+Action's should always include at least one response representing the HTTP response message sent back in response to a HTTP request. The response should always include a [status code][] and possibly a body.
```

Comment:
Does "should always" convey a MUST or a SHOULD? I would sync up this language with the spec. If MUST, use "Actions must include..." If SHOULD, "Actions should include..."

There are too many responses here as well (as noted by @zdne). I would even reword "in response to" to not have "response" in it.

Lastly, you have another "should always" in the last sentence. Sync this up with spec language, too.